ql/termstructures/yield/fittedbonddiscountcurve.hpp

```diff
@@ -197,24 +201,32 @@ namespace QuantLib {
         Real minimumCostValue() const;
         //! clone of the current object
         virtual std::auto_ptr<FittingMethod> clone() const = 0;
-		//! return whether there is a constraint at zero
-		bool constrainAtZero() const;
-		//! return weights being used
-		Array weights() const;
-		//! return optimization method being used
-		boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
-		//! open discountFunction to public
-		DiscountFactor discount(const Array& x, Time t) const;
+        //! return whether there is a constraint at zero
+        bool constrainAtZero() const;
+        //! return weights being used
+        Array weights() const;
+        //! return l2 penalties being used
+        Array l2() const;
+        //! return optimization method being used
+        boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
+        //! open discountFunction to public
+        DiscountFactor discount(const Array& x, Time t) const;
+        //! open discountFunction to public
+        Array gradients(const Array& x, Time t) const;
```

Comment:
I think the `discount`/`discountFunction` division is a historical artifact. It doesn't make a lot of sense to have a protected method and a public one forwarding directly to it.
I'd rather have just a public `gradient` method.